### PR TITLE
Uses the expiry interval from connect properties

### PR DESCRIPTION
**Pull request by andsel**
- Retrieve the session expiry interval from Connect property SESSION_EXPIRY_INTERVAL
- Added unit test to verify that CONNECT's SESSION_EXPIRY_INTERVAL property is used to set session expiration timeout in seconds
- Fallback to global expiry if one is not provided (if session is not clean)

### CODE REVIEW

File: broker/src/main/java/io/moquette/broker/SessionRegistry.java [hunk 1]
```diff
@@ -298,9 +299,22 @@ private Session createNewSession(MqttConnectMessage msg, String clientId) {
...
         final MqttVersion mqttVersion = Utils.versionFromConnect(msg);
+        if (mqttVersion != MqttVersion.MQTT_5) {
+            // in MQTT3 cleanSession = true means expiryInterval=0 else infinite
+            expiryInterval = clean ? 0 : globalExpirySeconds;
+        } else {
+            final MqttProperties.MqttProperty<Integer> expiryIntervalProperty =
+                (MqttProperties.MqttProperty<Integer>) msg.variableHeader().properties()
+                    .getProperty(MqttProperties.MqttPropertyType.SESSION_EXPIRY_INTERVAL.value());
+            if (expiryIntervalProperty != null) {
+                expiryInterval = expiryIntervalProperty.value();
```
Comment (hylkevds) [inline broker/src/main/java/io/moquette/broker/SessionRegistry.java]: For server stability it is probably a good idea to allow server operators to set a maximum allowed SESSION_EXPIRY_INTERVAL. For public, high-volume servers it is simply not feasible to allow random clients to create (effectively) non-expiring sessions, and we all know clients can't be trusted to properly set their intervals :)

Comment (andsel) [reply broker/src/main/java/io/moquette/broker/SessionRegistry.java]: Good point, some cap value configurable by settings :+1

Comment (andsel) [reply broker/src/main/java/io/moquette/broker/SessionRegistry.java]: Solved with 945ac33385fc7aa5889161d80dace1b2ee9a14e4

**Review by hylkevds — APPROVED** [review]
Looks good.

**Pull request merged**